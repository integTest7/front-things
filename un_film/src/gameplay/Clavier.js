import React from 'react'
import PropTypes from 'prop-types'

import './Clavier.scss'

const Clavier = ({ letter, onClick }) => (
    <div className="btn-custom">
        <button className={`letter ${letter}`} onClick={() => onClick(letter)}>
            { letter }
        </button>
    </div>
)

Clavier.propTypes = {
    letter: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired
}

export default Clavier