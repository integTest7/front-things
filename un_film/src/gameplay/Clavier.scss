.un-pendu {
    .letters-container {
        width: 94%;
        justify-content: center;
        background-color:#CCC;
        padding:0.4rem;
        border-radius: 0.4rem;
        
        .btn-custom {
            height: 2.6rem;
            width: 2.6rem;
            font-size: 1.2rem;
            color: #666666;
            margin: 0.2rem;
            display: flex;
            background-color: #FFFFFF;
            border-radius: 0.4rem;
            
            .letter {
                border:none;
                background: none;
                width:100%;
                height:100%;
                color:inherit;
                font-size:inherit;
                cursor:pointer;
                outline:none;
                
                &[disabled] {
                    background-color:#666666;
                    color: #333;
                }
            }
        }
    }
}

/*****************************************/
/*******        Media queries      *******/
/*****************************************/

/* Responsive */

@media (min-width: 768px) {
    .un-pendu {
        .letters-container {
            margin: 0 auto!important;
        }
    }
}