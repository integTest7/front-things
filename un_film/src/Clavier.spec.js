import React from 'react'
import { expect } from 'chai'
import { mount, shallow } from 'enzyme'
import sinon from 'sinon'

import Clavier from './gameplay/Clavier'

describe('<Clavier />', () => {
    it('should triggger its `onClick` prop when clicked', () => {
        const tuClick = sinon.spy()
        const wrapper = shallow(
            <Clavier letter="M" onClick={tuClick} />
        )      
    })
    
    it('should do a snapshot', () => {
        const tuClick = sinon.spy()
        const wrapper = shallow(
            <Clavier letter="M" onClick={tuClick} />
        )
        
        expect(wrapper).to.matchSnapshot()
    })
    
})