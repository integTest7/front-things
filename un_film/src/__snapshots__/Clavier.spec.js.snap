// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clavier /> should do a snapshot 1`] = `
<div
  className="btn-custom"
>
  <button
    className="letter M"
    onClick={[Function]}
  >
    M
  </button>
</div>
`;
